pkg/controller/enterprisesearch/config.go

```diff
@@ -251,16 +253,32 @@ func associationConfig(c k8s.Client, ent entv1beta1.EnterpriseSearch) (*settings
 		return settings.NewCanonicalConfig(), nil
 	}
 
+	// origin of authenticated ent users setting changed starting 8.x
+	cfg := settings.MustCanonicalConfig(map[string]string{
+		"ent_search.auth.source": "elasticsearch-native",
+	})
+	ver, err := version.Parse(ent.Spec.Version)
+	if err != nil {
+		return nil, err
+	}
+	if ver.IsSameOrAfter(version.From(8, 0, 0)) {
```

Comment:
Should we add a unit test to validate that the config is updated for versions > 8 ?